src/ServiceControl.Monitoring/Timings/TimingsStore.cs

```diff
@@ -0,0 +1,154 @@
+﻿namespace ServiceControl.Monitoring.Timings
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Processing.RawData.NServiceBus.Metrics;
+
+    public abstract class TimingsStore
+    {
+        ConcurrentDictionary<string, ConcurrentDictionary<DateTime, MeasurementInterval>> timings = 
+            new ConcurrentDictionary<string, ConcurrentDictionary<DateTime, MeasurementInterval>>();
+
+        public void Store(string endpointName, LongValueOccurrences message, DateTime now)
+        {
+            var endpointData = timings.GetOrAdd(endpointName, new ConcurrentDictionary<DateTime, MeasurementInterval>());
```

Comment:
The MSIL will look something like
```
LOAD endpointname
LOAD new ConcurrentDictionary
CALL GetOrAdd
```
It needs to call the constructor in order to pass the object in to the `GetOrAdd` call. There should be an overload of `GetOrAdd` that takes a `Func` that can delay that.